sphinxgallery/gen_rst.py

```diff
@@ -445,7 +445,7 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf, plot_gallery):
     Returns the set of functions/classes imported in the example.
     """
     base_image_name = os.path.splitext(fname)[0]
-    image_fname = '%s_%%03d.png' % base_image_name
+    image_fname = 'sphx_glr_%s_%%03d.png' % base_image_name
```

Comment:
> Another thought is if this whole hack of cleaning the images has to be done in the gallery scripts. I see nilearn and scikit-learn(https://github.com/scikit-learn/scikit-learn/blob/master/doc/Makefile#L40) clean the _build/html/_images with an instruction from the Makefile every time one uses make html, which seems more robust as it is done before sphinx starts doing anything. I wonder if it then would be best to just remove this hack from the gallery script and write in the Sphinx-gallery documentation to users to insert the extra instruction in their makefile.

It is no good option as it over deletes images that sphinx wont rebuild. If you are using pngmath instead of mathjax or if the matplotlib plot directive notices no change in the scripts it will assume the images still resides in `_build/html/_images` and it wont be copied again.

One has to keep track the images generated by sphinx-gallery but not where we save them but in the target directory of sphinx after it is renamed. A longer term fix would be to use the same procedure sphinx has to track images and only copy the ones that changed and not the whole gallery on every build.